app/models/vm_cloud_reconfigure_task.rb

```diff
@@ -0,0 +1,45 @@
+class VmCloudReconfigureTask < MiqRequestTask
+  alias_attribute :vm, :source
+
+  validate :validate_request_type, :validate_state
+
+  AUTOMATE_DRIVES = false
+
+  def self.base_model
+    VmCloudReconfigureTask
+  end
+
+  def self.get_description(req_obj)
+    name = nil
+    if req_obj.source.nil?
+      # Single source has not been selected yet
+      if req_obj.options[:src_ids].length == 1
+        v = Vm.find_by(:id => req_obj.options[:src_ids].first)
+        name = v.nil? ? "" : v.name
+      else
+        name = "Multiple VMs"
+      end
+    else
+      name = req_obj.source.name
+    end
+
+    new_settings = []
+    new_settings << "Flavor: #{req_obj.options[:flavor_name]}"
+    "#{request_class::TASK_DESCRIPTION} for: #{name} - #{new_settings.join(", ")}"
+  end
+
+  def after_request_task_create
+    update(:description => get_description)
+  end
+
+  def do_request
+    _log.info("Reconfiguring VM #{vm.id}:#{vm.name} to flavor #{options[:flavor_name]}")
+    vm.resize(options[:instance_type])
```

Comment:
@sseago This `vm.resize` call, and possibly the log message above, should move into the `else` block below since you only want it to run when `AUTOMATE_DRIVES` is `false`